src/components/TransactionHistoryRows/TransactionMethod.tsx

```diff
@@ -17,84 +17,99 @@ import { Transfers } from './TransactionDetails/Transfers'
 import { TxGrid } from './TransactionDetails/TxGrid'
 import { TransactionGenericRow } from './TransactionGenericRow'
 import type { TransactionRowProps } from './TransactionRow'
-import { AssetTypes, getTxMetadataWithAssetId, parseRelevantAssetFromTx } from './utils'
+import { getDisplayTransfers, getTxMetadataWithAssetId } from './utils'
 
-export const TransactionContract = ({
+export const TransactionMethod = ({
   txDetails,
   showDateAndGuide,
   compactMode,
   isOpen,
   toggleOpen,
   parentWidth,
 }: TransactionRowProps) => {
-  let assets = []
-  const txMetadata = useMemo(() => getTxMetadataWithAssetId(txDetails.tx.data), [txDetails.tx.data])
-  if (txDetails.sellAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Source))
-  if (txDetails.buyAsset) assets.push(parseRelevantAssetFromTx(txDetails, AssetTypes.Destination))
   const translate = useTranslate()
-  const isReceive = txDetails.tradeTx?.type === TransferType.Receive
-  const interactsWithWithdrawMethod = txDetails.tx.data?.method === ContractMethod.Withdraw
-  const isSend = txDetails.tradeTx?.type === TransferType.Send
-  const i18n =
-    isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
-  const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
+  const txMetadata = useMemo(() => txDetails.tx.data!, [txDetails.tx.data]) // we are guaranteed to have had metadata to render this component
+  const txMetadataWithAssetId = useMemo(() => getTxMetadataWithAssetId(txMetadata), [txMetadata])
 
-  const isRevoke = useMemo(
-    () =>
-      Boolean(
-        txMetadata?.method === 'approve' && txMetadata.value && bn(txMetadata.value).isZero(),
-      ),
-    [txMetadata],
+  const asset = useAppSelector(state =>
+    selectAssetById(state, txMetadataWithAssetId?.assetId ?? ''),
+  )
+
+  const displayTransfers = useMemo(
+    () => getDisplayTransfers(txDetails.transfers, [TransferType.Send, TransferType.Receive]),
+    [txDetails.transfers],
   )
+
   const titlePrefix = translate(
-    txDetails.tx.data?.parser
-      ? `transactionRow.parser.${txDetails.tx.data?.parser}.${isRevoke ? 'revoke' : i18n}`
+    txMetadata.parser
+      ? `transactionRow.parser.${txMetadata.parser}.${txMetadata.method}`
       : 'transactionRow.unknown',
   )
 
   // TODO(gomes): translation - we will need to revamp the prefix-suffix logic
   // to accomodate for different languages and their syntax
-  const titleSuffix = isRevoke ? ' approval' : ''
+  const titleSuffix = txMetadata.method === 'revoke' ? ' approval' : ''
 
-  const asset = useAppSelector(state =>
-    selectAssetById(state, getTxMetadataWithAssetId(txDetails.tx.data)?.assetId ?? ''),
-  )
-
-  const symbol = asset?.symbol ?? ''
+  const symbol = asset?.symbol
   const title = symbol ? `${titlePrefix} ${symbol}${titleSuffix}` : titlePrefix
 
+  const type = (() => {
```

Comment:
This eefee will be ran every render no matter whether or not dependencies change actually require it to be recomputed, can we either:

a. wrap it in a `useMemo`
b. abstract it into its `(method: string | undefined) => TransferType | Method | ""` pure function?

Note that these aren't mutually exclusive, the former is a requested change for memoization (not really a performance burden in this case, but something something consistency), and the latter, suggest change, could be consumed within said `useMemo()`'d expression